server/src/main/java/io/crate/metadata/view/ViewInfoFactory.java

```diff
@@ -79,14 +80,12 @@ public ViewInfo create(RelationName ident, ClusterState state) {
             }
             columns = collectedColumns;
         } catch (Exception e) {
-            if (e instanceof ResourceUnknownException) {
-                // Return ViewInfo with no columns in case the statement could not be analyzed,
-                // because the underlying table of the view could not be found.
-                columns = Collections.emptyList();
-            } else {
-                throw e;
-            }
+            // Statement could not be analyzed, because the referenced table either not found
+            // or has been updated and view definition became incompatible with the new schema (https://github.com/crate/crate/issues/14377).
+            columns = Collections.emptyList();
+            analyzeError = true;
         }
-        return new ViewInfo(ident, view.stmt(), columns, view.owner());
+        String viewDefinition = analyzeError ? String.format(Locale.ENGLISH, "/* Corrupted view, needs fix */ %s", view.stmt()) : view.stmt();
```

Comment:
How about adding a line break to make the comment even more visible.
```suggestion
        String viewDefinition = analyzeError ? String.format(Locale.ENGLISH, "/* Corrupted view, needs fix */\n%s", view.stmt()) : view.stmt();
```